kedro/io/core.py

```diff
@@ -406,7 +406,21 @@ def parse_dataset_definition(
                 class_obj = tmp
                 break
         else:
-            raise DatasetError(f"Class '{dataset_type}' not found, is this a typo?")
+            hint = ""
+            if "DataSet" in dataset_type:
+                hint = (
+                    "Hint: If you are trying to use a dataset from `kedro-datasets`>=2.0.0, "
+                    "make sure that the dataset name uses the `Dataset` spelling instead of `DataSet`."
+                )
+            else:
+                hint = (
+                    "Hint: If you are trying to use a dataset from `kedro-datasets`, "
+                    "make sure that the package is installed in your current environment. "
+                    "You can do so by running `pip install kedro-datasets`."
```

Comment:
```suggestion
                    "You can do so by running `pip install kedro-datasets` or `pip install kedro-datasets[polars] for a specific back-end`."
```

wdyt?

Reply:
I like @datajoely 's suggestion, but I wouldn't use the wording "back-end" and just say for a specific dataset.

Reply:
I added this with a slightly different wording